code/modules/clothing/gloves/miscellaneous.dm

```diff
@@ -99,27 +99,29 @@
 	item_state = "wgloves"
 	var/range = 3
 
-/obj/item/clothing/gloves/color/white/magic/attackby(obj/item/W, mob/user, params)
+/obj/item/clothing/gloves/color/white/magic/attackby(obj/item/attacking_item, mob/user, params)
 	. = ..()
-	if(istype(W, /obj/item/upgradewand))
-		var/obj/item/upgradewand/wand = W
+	if(istype(attacking_item, /obj/item/upgradewand))
+		var/obj/item/upgradewand/wand = attacking_item
 		if(!wand.used && range == initial(range))
 			wand.used = TRUE
 			range = 6
 			to_chat(user, "<span_class='notice'>You upgrade the [src] with the [wand].</span>")
 			playsound(user, 'sound/weapons/emitter2.ogg', 25, 1, -1)
 
-/obj/item/clothing/gloves/color/white/magic/Touch(atom/A, proximity)
-	var/mob/living/M = loc
-	if(get_dist(A, M) <= 1)
-		return 0
-	if(M in viewers(range, A))
-		M.visible_message("<span_class ='danger'>[M] waves their hands at [A]</span>", "<span_class ='notice'>You begin manipulating [A].</span>")
-		new	/obj/effect/temp_visual/telegloves(A.loc)
-		M.changeNext_move(CLICK_CD_MELEE)
-		if(do_after_mob(M, A, 8))
-			new /obj/effect/temp_visual/telekinesis(M.loc)
-			playsound(M, 'sound/weapons/emitter2.ogg', 25, 1, -1)
-			A.attack_hand(M)
-			return 1
+/obj/item/clothing/gloves/color/white/magic/Touch(atom/target, proximity)
+	var/mob/living/user = loc
+	if(get_dist(target, user) <= 1)
+		return FALSE
+	if(istype(target, /obj/structure/pool_ladder))
```

Comment:
This should probably just be any structures in general to prevent other issues, instead of just picking at each and every possible exploit.

Reply:
Consider it done